internal/jobs/pubkey_upload_aws_job.go

```diff
@@ -108,12 +121,12 @@ func HandlePubkeyUploadAWS(ctx context.Context, job dejq.Job) error {
 		return fmt.Errorf("cannot upload aws pubkey: %w", err)
 	}
 
-	// mark the reservation as finished
+	// update the reservation status, but keep it unfinished yet
 	rDao, err := dao.GetReservationDao(ctx)
 	if err != nil {
 		return fmt.Errorf("cannot get reservation DAO: %w", err)
 	}
-	err = rDao.UpdateStatus(ctx, args.ReservationID, "Finished")
+	err = rDao.UpdateStatus(ctx, args.ReservationID, "Key uploaded")
```

Comment:
This is why I think we definitely need to improve on status recognition - only Status field is not enough. We do have Success flag but it was unused until this patch.